### ISSUE TITLE
Class instance method does not determine type of self

### ISSUE THREAD

**Issue by BurningLights** [microsoft/pyright#155]
**Is your feature request related to a problem? Please describe.**
When creating a method in a class, pyright  considers the type of the first parameter 'self' as Unknown. Example:
```
class Foo:
    def __init__(self, bar):
         self.bar = bar
```

**Describe the solution you'd like**
It would be great if pyright automatically determined the type of the self parameter for methods to be the type of the class.

**Additional context**
Screenshot of current behavior in Visual Studio Code:
![pyright suggestion](https://user-images.githubusercontent.com/14267553/59932551-6c143b00-9415-11e9-9782-5f539fb16e2c.png)

If this change were implemented, the popup would say:
```
class Foo
(parameter) self: Foo
```

**Comment by erictraut** [comment]
Thanks for reporting the bug. Pyright was already correctly inferring the type of "self" in these cases, but it contained a bug that prevented the correct inferred type from being reported when you hovered over the parameter in VS Code. This will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.0.41, which I just published. Thanks again for the bug report!
